File: C++/level2/피보나치_수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int first = 0;
    int second = 1;
    int temp;

    for(int i=0; i<n-1; i++){
        temp = first;
        first = second;
        second = (first+temp)%1234567;
    }
    
    return second;
}

File: JavaScript/level2/피보나치_수.js
function solution(n) {
    let first = 0;
    let second = 1;
    let temp;
    
    if(n === 1)
        return 1;
    
    for(let i=1; i<n; i++){
        temp = second;
        second = (first + second)%1234567;
        first = temp;
    }
    
    return second;
}